opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -3205,6 +3214,7 @@ namespace Opm
             duneD_[seg][outlet_segment_index][SPres][pv_idx] = -outlet_pressure.derivative(pv_idx + numEq);
 
         }
+        duneDSolver_.reset();
```

Comment:
I do not think we need to keep doing this in multiple location. Maybe only do it at the end of function `assembleWellEqWithoutIteration`? It is as far as what I can see now. I can miss something.

Reply:
Or can be easier to understand from code perspective, do the reset after the matrix D is cleared. 
```
2512 
2513         duneD_ = 0.0;
```

Reply:
Will try.